allennlp/models/encoder_decoders/wikitables_semantic_parser.py

```diff
@@ -450,25 +448,28 @@ def _get_linking_probabilities(worlds: List[WikiTablesWorld],
                                                  dim=1,
                                                  index=Variable(linking_scores.data.new(row_type_index)).long())
 
-            # (num_question_tokens, num_entities_per_type)
+            # (num_question_tokens, num_entities_per_type + 1)
             cell_type_scores[:, 0] = 0
             row_type_scores[:, 0] = 0
 
-            probabilities_cell = torch.nn.functional.softmax(cell_type_scores)
-            probabilities_row = torch.nn.functional.softmax(row_type_scores)
+            probabilities_cell = torch.nn.functional.softmax(cell_type_scores, dim=1)
+            probabilities_row = torch.nn.functional.softmax(row_type_scores, dim=1)
+
+            # NOTE: The way that we're doing this here relies on the fact that entities are
+            # implicitly sorted by their types when we sort them by name, and that "fb:cell" comes
+            # before "fb:row".  This is not a great assumption, and could easily break later, but
+            # it should work for now.
+            to_cat = [probabilities_cell[:, 1:], probabilities_row[:, 1:]]
```

Comment:
minor: maybe to_cat could be named probabilities_all_types and num_kept_entities to num_entities_in_batch

Reply:
Done.